libafl/src/corpus/cached.rs

```diff
@@ -20,27 +20,13 @@ use crate::{
 /// The eviction policy is FIFO.
 #[cfg(feature = "std")]
 #[derive(Default, Serialize, Deserialize, Clone, Debug)]
-#[serde(bound = "I: serde::de::DeserializeOwned")]
-pub struct CachedOnDiskCorpus<I>
-where
-    I: Input,
-{
+pub struct CachedOnDiskCorpus<I> {
     inner: InMemoryOnDiskCorpus<I>,
     cached_indexes: RefCell<VecDeque<CorpusId>>,
     cache_max_len: usize,
 }
 
-impl<I> UsesInput for CachedOnDiskCorpus<I>
```

Comment:
Why would we want a corpus of non-inputs?

Reply:
I have entirely removed Uses{Input,State} because it overconstrains usage. We have better ways to implement this by "moving down" the constraint to the relevant callsite.

Reply:
Also, you'll notice that the implementations of Corpus still constraint I: Input :P